test/api/custom_layer.test.cpp

```diff
@@ -96,14 +96,16 @@ TEST(CustomLayer, Basic) {
     map.getStyle().addLayer(std::make_unique<CustomLayer>(
         "custom",
         [] (void* context) {
-            reinterpret_cast<TestLayer*>(context)->initialize();
+          reinterpret_cast<TestLayer*>(context)->initialize();
         },
         [] (void* context, const CustomLayerRenderParameters&) {
-            reinterpret_cast<TestLayer*>(context)->render();
+          reinterpret_cast<TestLayer*>(context)->render();
         },
         [] (void* context) {
-            delete reinterpret_cast<TestLayer*>(context);
-        }, new TestLayer()));
+          delete reinterpret_cast<TestLayer*>(context);
+        },
+        nullptr,
```

Comment:
This test needs to be changed to delete the `context` ptr in the `deallocationFn` callback instead of the `deinitializeFn` callback